src/app/wallets/PeraConnect.tsx

```diff
@@ -0,0 +1,97 @@
+import algosdk from "algosdk";
+import { PeraWalletConnect } from "@perawallet/connect";
+import { WalletConnectorProps, WalletName } from "./types";
+import { useCallback, useMemo } from "react";
+import { useTransaction } from "~/hooks/useTransaction";
+import { useToast } from "~/components/ui/use-toast";
+import { Avatar, AvatarFallback, AvatarImage } from "~/components/ui/avatar";
+
+export const PeraConnect: React.FC<WalletConnectorProps> = ({
+  setWalletAddresses,
+  transactionPayload,
+}) => {
+  const { toast } = useToast();
+  const { setSignedTransaction } = useTransaction();
+
+  const getAddresses = useCallback(async () => {
+    const peraWallet = new PeraWalletConnect();
```

Comment:
I'm just not sure if we want to init this here or into a singleton, but we can keep it like this since we plan to rework this part